File: source/xamarin/Xamarin.Droid.Sample/ShowCardFragment.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using AdaptiveCards.Rendering.Xamarin.Android.ObjectModel;
using AdaptiveCards.Rendering.Xamarin.Android.Renderer;
using AdaptiveCards.Rendering.Xamarin.Android.Renderer.ActionHandler;
using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using Android.Support.V4.App;

namespace AdaptiveCards.Rendering.Xamarin.Android.Sample
{
    class ShowCardFragment : global::Android.Support.V4.App.DialogFragment
    {

        public void initialize(Context _context, global::Android.Support.V4.App.FragmentManager _fragmentManager, ShowCardAction showCardAction, ICardActionHandler cardActionHandler, HostConfig hostConfig)
        {
            context = _context;
            fragmentManager = _fragmentManager;
            ShowCardAction = showCardAction;
            CardActionHandler = cardActionHandler;
            HostConfig = hostConfig;
        }

        public override View OnCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
        {
            Dialog.Window.SetSoftInputMode(SoftInput.AdjustResize);

            View v = inflater.Inflate(Resource.Layout.popup_fragment, container);

            RenderedAdaptiveCard renderedCard = AdaptiveCardRenderer.Instance.Render(context, fragmentManager, ShowCardAction.Card, CardActionHandler, HostConfig);

            ViewGroup viewGroup = (ViewGroup)renderedCard.View;
            Dialog.SetTitle(ShowCardAction.Title);

            ViewGroup insertPoint = (ViewGroup)v.FindViewById(Resource.Id.popup_fragment);
            insertPoint.AddView(viewGroup);
            v.Focusable = true;
            return v;
        }

        public override void OnResume()
        {
            base.OnResume();

            Dialog dialog = Dialog;
            if (dialog != null)
            {
                dialog.Window.SetLayout(ViewGroup.LayoutParams.MatchParent, ViewGroup.LayoutParams.WrapContent);
            }
        }

        private Context context;
        private global::Android.Support.V4.App.FragmentManager fragmentManager;
        private ShowCardAction ShowCardAction;
        private ICardActionHandler CardActionHandler;
        private HostConfig HostConfig;
    }
}

File: source/android/mobilechatapp/src/main/java/com/example/mobilechatapp/ShowCardFragment.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
package com.example.mobilechatapp;

import android.app.Dialog;
import android.content.Context;
import androidx.fragment.app.DialogFragment;

import android.os.Bundle;
import androidx.fragment.app.FragmentManager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.WindowManager;

import io.adaptivecards.objectmodel.HostConfig;
import io.adaptivecards.objectmodel.ShowCardAction;
import io.adaptivecards.renderer.AdaptiveCardRenderer;
import io.adaptivecards.renderer.RenderedAdaptiveCard;
import io.adaptivecards.renderer.actionhandler.ICardActionHandler;

public class ShowCardFragment extends DialogFragment
{
    public void initialize(Context context, FragmentManager fragmentManager, ShowCardAction showCardAction, ICardActionHandler cardActionHandler, HostConfig hostConfig)
    {
        m_context = context;
        m_fragmentManager = fragmentManager;
        m_showCardAction = showCardAction;
        m_cardActionHandler = cardActionHandler;
        m_hostConfig = hostConfig;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
    {
        getDialog().getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_RESIZE);

        View v = inflater.inflate(R.layout.popup_fragment, container);

        RenderedAdaptiveCard renderedCard = AdaptiveCardRenderer.getInstance().render(m_context, m_fragmentManager, m_showCardAction.GetCard(), m_cardActionHandler, m_hostConfig);

        ViewGroup viewGroup = (ViewGroup) renderedCard.getView() ;
        getDialog().setTitle(m_showCardAction.GetTitle());

        ViewGroup insertPoint = (ViewGroup) v.findViewById(R.id.popup_fragment);
        insertPoint.addView(viewGroup);
        v.setFocusable(true);
        return v;
    }

    @Override
    public void onResume()
    {
        super.onResume();

        Dialog dialog = getDialog();
        if (dialog != null) {
            dialog.getWindow().setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);
        }

    }

    private Context m_context;
    private FragmentManager m_fragmentManager;
    private ShowCardAction m_showCardAction;
    private ICardActionHandler m_cardActionHandler;
    private HostConfig m_hostConfig;
}
